designer/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/scenariodetails/ScenarioWithDetails.scala

```diff
@@ -0,0 +1,62 @@
+package pl.touk.nussknacker.restmodel.scenariodetails
+
+import io.circe.generic.JsonCodec
+import pl.touk.nussknacker.engine.api.deployment.{ProcessAction, ProcessState}
+import pl.touk.nussknacker.engine.api.process.{ProcessId => ApiProcessId, ProcessIdWithName, ProcessName, VersionId}
+import pl.touk.nussknacker.restmodel.displayedgraph.{DisplayableProcess, ValidatedDisplayableProcess}
+import pl.touk.nussknacker.restmodel.process.ProcessingType
+import pl.touk.nussknacker.restmodel.validation.ValidationResults
+
+import java.time.Instant
+
+@JsonCodec
+final case class ScenarioWithDetails(
+    id: String,
+    name: ProcessName,
+    processId: ApiProcessId,
+    processVersionId: VersionId,
+    isLatestVersion: Boolean,
+    description: Option[String],
+    isArchived: Boolean,
+    isFragment: Boolean,
+    processingType: ProcessingType,
+    processCategory: String,
+    modificationDate: Instant,
+    modifiedAt: Instant,
+    modifiedBy: String,
+    createdAt: Instant,
+    createdBy: String,
+    tags: Option[List[String]],
+    lastDeployedAction: Option[ProcessAction],
+    lastStateAction: Option[ProcessAction],
+    lastAction: Option[ProcessAction],
+    // TODO: move things like processingType, category and validationResult on the root level and rename json to scenarioGraph
+    json: Option[ValidatedDisplayableProcess],
+    history: Option[List[ScenarioVersion]],
+    modelVersion: Option[Int],
+    state: Option[ProcessState]
+) {
+
+  lazy val idWithName: ProcessIdWithName = ProcessIdWithName(processId, name)
+
+  def withScenarioGraphAndValidationResult(
+      scenarioWithValidationResult: ValidatedDisplayableProcess
+  ): ScenarioWithDetails = {
+    copy(json = Some(scenarioWithValidationResult))
+  }
+
+  def historyUnsafe: List[ScenarioVersion] = history.getOrElse(throw new IllegalStateException("Missing history"))
```

Comment:
`None` doesn't mean that we have no versions. It means that we didn't fetched it from db so we don't know what versions are.

Reply:
IMO None means that the scenario doesn't have any tags :/ As I understand you want to do something similar to json and fetch's type? So right now it's consistent, because we have two ways to express that :/

Reply:
It is very interesting what did you write :) None mean not defined / not specified / not present. In Java there is `ifPresent`, in scala `isDefined`.
Let's go through examples. Let's say that you have one tag "foo" and want to remove it. What do you expect? That from `Some(List("foo"))` you will suddenly have `None`. Where is the logic?
Regarding scenario shape it is because we decided to represent scenario as a functor, in theory it suppose to help us to easily go through various steps of transformation: Canonical -> Displayable -> Validated. In practice it was a mistake. We will clean this probably when we remove Canonical representation.

Reply:
I said it wrong..  I wanted to say that None means that in an object there is no property tags. This pattern with the funktor is okey for me. I just want to say that right now we have two way of express that our Scenario doesn't include some property. Once there are tags, history other it's the json.